tools/install.sh

```diff
@@ -43,19 +62,25 @@ mv -f ~/.zshrc-omztemp ~/.zshrc
 
 TEST_CURRENT_SHELL=$(expr "$SHELL" : '.*/\(.*\)')
 if [ "$TEST_CURRENT_SHELL" != "zsh" ]; then
-    echo "\033[0;34mTime to change your default shell to zsh!\033[0m"
+    printf "${BLUE}Time to change your default shell to zsh!${NORMAL}\n"
     chsh -s $(grep /zsh$ /etc/shells | tail -1)
 fi
 unset TEST_CURRENT_SHELL
 
-echo "\033[0;32m"'         __                                     __   '"\033[0m"
-echo "\033[0;32m"'  ____  / /_     ____ ___  __  __   ____  _____/ /_  '"\033[0m"
-echo "\033[0;32m"' / __ \/ __ \   / __ `__ \/ / / /  /_  / / ___/ __ \ '"\033[0m"
-echo "\033[0;32m"'/ /_/ / / / /  / / / / / / /_/ /    / /_(__  ) / / / '"\033[0m"
-echo "\033[0;32m"'\____/_/ /_/  /_/ /_/ /_/\__, /    /___/____/_/ /_/  '"\033[0m"
-echo "\033[0;32m"'                        /____/                       ....is now installed!'"\033[0m"
-echo "\n\n \033[0;32mPlease look over the ~/.zshrc file to select plugins, themes, and options.\033[0m"
-echo "\n\n \033[0;32mp.s. Follow us at http://twitter.com/ohmyzsh.\033[0m"
-echo "\n\n \033[0;32mp.p.s. Get stickers and t-shirts at http://shop.planetargon.com.\033[0m"
+printf "${GREEN}"
+echo '         __                                     __   '
+echo '  ____  / /_     ____ ___  __  __   ____  _____/ /_  '
+echo ' / __ \/ __ \   / __ `__ \/ / / /  /_  / / ___/ __ \ '
+echo '/ /_/ / / / /  / / / / / / /_/ /    / /_(__  ) / / / '
+echo '\____/_/ /_/  /_/ /_/ /_/\__, /    /___/____/_/ /_/  '
+echo '                        /____/                       ....is now installed!'
+echo ''
+echo ''
+echo 'Please look over the ~/.zshrc file to select plugins, themes, and options.'
+echo ''
+echo 'p.s. Follow us at https://twitter.com/ohmyzsh.'
+echo ''
+echo 'p.p.s. Get stickers and t-shirts at http://shop.planetargon.com.'
+echo ''
+printf "${NORMAL}"
 env zsh
```

Comment:
The old `. ~/.zshrc` wasn't actually wrong in that manner with the old method, just confusing. In the case of `curl | sh`, `sh` doesn't process the entire input file as a script at once; it reads and evaluates it line by line. When that `| sh` spawns a new `zsh`, that `zsh` session *takes over* stdin (which is attached to `curl |` and not the user's tty/keyboard) and reads the rest of the input. So all the lines after `env zsh` are actually processed by that `zsh` subshell. But then `zsh` runs out of input and exits, returning control to `sh`, so the user never got to actually work in that new `zsh` session. I think people just didn't realize this was happening because the installer was commonly run from a `zsh` session anyway, at least by developers.

This is the same reason that `chsh -s` was failing on systems where it read from `stdin` instead of directly from `/dev/tty`: it would consume the next line of `install.sh` instead of new input from the keyboard, and use that as the password.

Here's a simple script to verify it. Save this as `foo.sh`.
```
echo Hello $$ $0
env zsh
echo Another hello $$ $0
```

I get the following.
```
eilonwy:~ janke$ cat foo.sh | sh
Hello 3903 sh
Another hello 3904 zsh
eilonwy:~ janke$
```

Anyway, moot point, since we're not using `curl | sh` any more.

###   What to do now   ###

Oh yeah. You're right. I didn't see that `exec` would be replacing the `sh -c` subshell that was running `install.sh`, and not the interactive shell that you ran it from. Sorry. Maybe I've been thinking in terms of shell functions too much lately.

I still think it's a bit of an annoyance and would personally prefer just a printed "restart your terminal to get OMZ", because I wouldn't expect an installer to go right ahead and run a program, and you've still got that original non-OMZ shell parent session running, which seems a little messy. But it's not as bad as I said in my previous comment. I'm fine with whichever way you guys want to do it.